pycontrails/core/datalib.py

```diff
@@ -505,6 +526,19 @@ def open_metdataset(
         :func:`xarray.open_mfdataset`
         """
 
+    @abc.abstractmethod
+    def set_met_source_metadata(self, ds: xr.Dataset | MetDataset) -> None:
```

Comment:
Why not call this `set_attrs` or `set_metadata`? A little less verbose

Reply:
c9c2957fc4659033c477a6745c5c7bc11f12a531